include/Mac.h

```diff
@@ -40,6 +40,12 @@ class Mac : public GenericHashEntry {
     char * dhcp; /* Extracted from DHCP dissection */
   } names;
 
+  struct{
+    u_int32_t as_server;
+    u_int32_t as_client;
+  }talkers; /*local talkers*/
+  DeviceTypeCounters dev_t_counters = {0,0,0,0,0,0,0,0,0,0,0,0,0};/*type of the devices this mac talked to*/
```

Comment:
wasting a huge amount of memory for every Mac... not necessary. You have the matrix, you can keep everything there and obtain the counters simply by navigating  the matrix you already have.